src/opm/output/eclipse/AggregateWellData.cpp

```diff
@@ -0,0 +1,773 @@
+                WV::NumWindows{ numWells(inteHead) },
+                WV::WindowSize{ entriesPerWell(inteHead) }
+            };
+        }
+
+        int groupIndex(const std::string&              grpName,
+                       const std::vector<std::string>& groupNames,
+                       const int                       maxGroups)
+        {
+            if (grpName == "FIELD") {
+                // Not really supposed to happen since wells are
+                // not supposed to be parented dirctly to FIELD.
+                return maxGroups + 1;
+            }
+
+            auto b = std::begin(groupNames);
+            auto e = std::end  (groupNames);
+            auto i = std::find(b, e, grpName);
+
+            if (i == e) {
+                // Not really supposed to happen since wells are
+                // not supposed to be parented dirctly to FIELD.
+                return maxGroups + 1;
+            }
+
+            // One-based indexing.
+            return std::distance(b, i) + 1;
+        }
+
+        int wellType(const Opm::Well&  well,
+                     const std::size_t sim_step)
+        {
+            if (well.isProducer(sim_step)) {
+                return 1;      // Producer flag
+            }
+
+            using IType = ::Opm::WellInjector::TypeEnum;
+
+            const auto itype = well
+                .getInjectionProperties(sim_step).injectorType;
+
+            switch (itype) {
+            case IType::OIL:   return 2; // Oil Injector
+            case IType::WATER: return 3; // Water Injector
+            case IType::GAS:   return 4; // Gas Injector
+            default:           return 0; // Undefined
+            }
+        }
+
+        int wellVFPTab(const Opm::Well&  well,
+                       const std::size_t sim_step)
+        {
+            if (well.isInjector(sim_step)) {
+                return well.getInjectionProperties(sim_step).VFPTableNumber;
+            }
+
+            return well.getProductionProperties(sim_step).VFPTableNumber;
+        }
+
+        int ctrlMode(const Opm::Well&  well,
+                     const std::size_t sim_step)
+        {
+            {
+                const auto stat = well.getStatus(sim_step);
+
+                using WStat = ::Opm::WellCommon::StatusEnum;
+
+                if ((stat == WStat::SHUT) || (stat == WStat::STOP)) {
+                    return 0;
+                }
+            }
+
+            if (well.isInjector(sim_step)) {
+                const auto& prop = well
+                    .getInjectionProperties(sim_step);
+
+                const auto wmctl = prop.controlMode;
+                const auto wtype = prop.injectorType;
+
+                using CMode = ::Opm::WellInjector::ControlModeEnum;
+                using WType = ::Opm::WellInjector::TypeEnum;
+
+                switch (wmctl) {
+                case CMode::RATE: {
+                    switch (wtype) {
+                    case WType::OIL:   return 1; // ORAT
```

Comment:
The integers used for the various control modes should be given symbolic names somewhere.